qiskit/pulse/channels.py

```diff
@@ -44,8 +45,9 @@ def __init__(self, index: int):
         Raises:
             PulseError: If ``index`` is not a nonnegative integer.
         """
-        if not isinstance(index, int) or index < 0:
-            raise PulseError('Channel index must be a nonnegative integer')
+        if not isinstance(index, ParameterExpression):
+            if not isinstance(index, int) or index < 0:
+                raise PulseError('Channel index must be a nonnegative integer')
```

Comment:
```suggestion
        if (not isinstance(index, ParameterExpression) 
                and not isinstance(index, int) or index < 0):
            raise PulseError('Channel index must be a nonnegative integer or a ParameterExpression.')
```
https://www.python.org/dev/peps/pep-0008/#multiline-if-statements

Reply:
I think it's more readable as is